tool/grind/utils.dart

```diff
@@ -62,3 +64,29 @@ ArchiveFile fileFromString(String path, String contents,
 ArchiveFile file(String target, String source, {bool executable: false}) =>
     fileFromBytes(target, new File(source).readAsBytesSync(),
         executable: executable);
+
+/// Ensure that the repository at [url] is cloned into the build directory and
+/// pointing to the latest master revision.
+///
+/// Returns the path to the repository.
+Future<String> cloneOrPull(String url) async {
+  var name = p.url.basename(url);
+  if (p.url.extension(name) == ".git") name = p.url.withoutExtension(name);
+
+  var path = p.join("build", name);
+
+  if (new Directory(p.join(path, '.git')).existsSync()) {
+    log("Updating $url");
+    await runAsync("git",
+        arguments: ["fetch", "origin"], workingDirectory: path);
+    await runAsync("git",
+        arguments: ["reset", "--hard", "origin/master"],
```

Comment:
This can be a bit destructive on your local changes, if you ran it by accident. Not only to the working tree but also to the current branch pointer. I wonder if it'd be worth a `git checkout origin/master` just to be safe, so it'd error out if you had local changes.

Since it's only used for other sass impl repos it's probably pretty safe even so, but just wondering if the extra level of `reset --hard` is necessary.

Reply:
`checkout` doesn't update the local `master` branch's pointer, which I worry could hypothetically mess up some infrastructure. These repositories are all in the `build/` directory anyway, which is generally expected to be destructively modified by grinder tasks, so I don't feel too bad about doing that here as well.